doc/api/modules.md

```diff
@@ -188,33 +188,50 @@ loaded by `require()` meets the following requirements:
      `"type": "commonjs"`, and `--experimental-detect-module` is enabled.
 
 `require()` will load the requested module as an ES Module, and return
-the module name space object. In this case it is similar to dynamic
+the module namespace object. In this case it is similar to dynamic
 `import()` but is run synchronously and returns the name space object
 directly.
 
+With the following ES Modules:
+
 ```mjs
-// point.mjs
+// distance.mjs
 export function distance(a, b) { return (b.x - a.x) ** 2 + (b.y - a.y) ** 2; }
+```
+
+```mjs
+// point.mjs
 class Point {
   constructor(x, y) { this.x = x; this.y = y; }
 }
 export default Point;
 ```
 
+A CommonJS module can load them with `require()` under `--experimental-detect-module`:
+
 ```cjs
-const required = require('./point.mjs');
+const distance = require('./distance.mjs');
+console.log(distance);
 // [Module: null prototype] {
-//   default: [class Point],
 //   distance: [Function: distance]
 // }
-console.log(required);
 
-(async () => {
-  const imported = await import('./point.mjs');
-  console.log(imported === required);  // true
-})();
+const point = require('./point.mjs');
+console.log(point);
+// [Module: null prototype] {
+//   default: [class Point],
+//   __esModule: true,
+// }
 ```
 
+For interoperability with existing tools that convert ES Modules into CommonJS,
+which could then load real ES Modules through `require()`, the returned namespace
+would contain a `__esModule: true` property if it has a `default` export so that
+consuming code generated by tools can recognize the default exports in real
+ES Modules. This property is experimental and can change in the future.
+CommonJS code that are not converted from ES Modules by tools should not
+rely on this property.
```

Comment:
I'm not sure I follow this sentence, could it rephrased? Something about "CommonJS code that are not converted from ES Modules" doesn't click. Maybe a more general statement?

(This could also just be me)

Reply:
It basically means only code that are generated by tools should not use this property.

Reply:
I'm reading it as the opposite. It (to me) sounds like if the code is generated by a tool, it *shouldn't* use the property.